#+STARTUP: latexpreview
#+TITLE: Genome Wide Association Study of Taste in ALSPAC
#+AUTHOR: Sam Neaves

* Set up


In order to insert images directly into the org file make sure you
have run the following emacs commands.

#+BEGIN_SRC lisp
(setq org-confirm-babel-evaluate nil)
(add-hook 'org-babel-after-execute-hook 'org-display-inline-images)   
(add-hook 'org-mode-hook 'org-display-inline-images)  

#+END_SRC


To get the var from ALSPAC on local machine:
#+BEGIN_SRC R :session *shell* :results output :tangle get_taste_pheno.R
  library(devtools)
  ### library(alspac is available from: https://github.com/explodecomputer/alspac)
  library(alspac)
  setDataDir("/Volumes/ALSPAC-Data/")
  data(current)
  data(useful)


  varnames <- c("fdfa041")
  vars <- findVars(varnames)
  results <- extractVars(vars)

  write.csv(as.data.frame(results),file="./pheno_taste.csv")

#+END_SRC

To run the R code as an interactive job on bc4, run the following:
#+NAME: ijob
#+PROPERTY: header-args :eval never-export
#+BEGIN_SRC bash
srun --nodes=1 --ntasks-per-node=1 --mem=10G --time=08:00:00 --pty bash -i
#+END_src


* Exploring Phenotype data
To load Phenotype into R on BC4
** Reading the file into R

#+BEGIN_SRC R :session *shell*<2> :results output :exports both

  raw_pheno <-read.csv('pheno_taste.csv')


#+END_SRC

#+RESULTS:
: 
: raw_pheno <-read.csv('pheno_taste.csv')

** Excluding people
Participants without taste var excluded


#+BEGIN_SRC R :session *shell*<2> :results output :exports both
  library(dplyr)
  dim(raw_pheno)


  pheno_with_taste_raw <-raw_pheno %>% filter(!is.na(fdfa041))

  dim(pheno_with_taste_raw)

  pheno_with_taste_raw2 <-pheno_with_taste_raw %>% filter(fdfa041>0)  

  dim(pheno_with_taste_raw2)
#+END_SRC

#+RESULTS:
#+begin_example
library(dplyr)
dim(raw_pheno)
[1] 15645    31


pheno_with_taste_raw <-raw_pheno %>% filter(!is.na(fdfa041))

dim(pheno_with_taste_raw)
[1] 7532   31

pheno_with_taste_raw2 <-pheno_with_taste_raw %>% filter(fdfa041>0)

dim(pheno_with_taste_raw2)
[1] 5150   31
#+end_example

** 

** Examining taste var 
*** summary

#+BEGIN_SRC R :session *shell*<2> :results output :exports both
  summary(pheno_with_taste_raw2$fdfa041)
#+END_SRC

#+RESULTS:
: summary(pheno_with_taste_raw2$fdfa041)
:    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
:   0.100   4.700   7.300   6.525   8.800  10.000


#+BEGIN_SRC R :file taste1.png :results output graphics :session *shell*<2> :exports both
library(ggplot2)
library(cowplot)

theme_set(theme_cowplot())
p1 <- ggplot(pheno_with_taste_raw2,aes(fdfa041)) +
geom_histogram(fill = "darkblue",
na.rm = TRUE)
print(p1)



#+END_SRC

#+RESULTS:
[[file:taste1.png]]


*** Writing to file

We need to create our new fam file that has the phenotype data.


#+BEGIN_SRC R :session *shell*<2> :results output :exports both
  fam_original <- read.table("../combined.fam")


  library(dplyr)

  new_fam_all = fam_original %>% left_join(pheno_with_taste_raw2,by = c("V1" = "alnqlet"))

  new_fam <- new_fam_all[c("V1","V2","V3","V4","V5","fdfa041")]
  new_fam[is.na(new_fam)] <- -9

  write.table(new_fam,"taste.fam", row.names=FALSE, col.names=FALSE, quote=FALSE)


#+END_SRC

** HRC data

We want to use the [[HRC data][HRC data]].
These need to be in the correct format for Gemma.

We are initially using plink format files.


* Running GWASes with Gemma

Todo this we are going to tangle a biomake file with rules for how to
execute Gemma, including the dependencies. The emacs command to
perform a tangle is C-c C-v t whilst point is in the code block.

We can then execute the recipes in the make file over slurm in compute
jobs.


** Aim compute the p-values to assess support for association between scar length and SNPs on chromosomes 1â€“22.

1. Run a basic association analysis in GEMMA.
2. Run an LMM-based association analysis in GEMMA.
3. Compare the two association analyses.

** Basic association

For the first basic association we do not have any covars and use the
gemma -lm 1 option which performs a Wald test.
Create the rule:

#+PROPERTY: header-args :eval never-export
#+BEGIN_SRC prolog :tangle makeprog
'./output/taste.assoc.txt',{QsubArgs = '--time=18:00:00'}<--
['taste.bed','taste.bim','taste.fam'],'../../../gemma -b taste -lm 1 -outdir . -o ./output/taste'.

'./output/taste.log.txt' <--['./output/taste.assoc.txt'],{true}.

[gemma_basic] <--['./output/taste.log.txt'].


#+END_SRC

Evoke the recipes as submitted jobs on bc4 with slurm.
#+BEGIN_SRC bash :results output

biomake -Q slurm gemma_basic

#+END_SRC

#+RESULTS:
: Submitting job: sbatch -o /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/out/taste.assoc.txt -e /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/err/taste.assoc.txt   --time=18:00:00  --parsable /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/script/taste.assoc.txt >/user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/job/taste.assoc.txt
: Submitting job: sbatch -o /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/out/taste.log.txt -e /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/err/taste.log.txt    --dependency=afterok:10733694 --parsable /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/script/taste.log.txt >/user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/job/taste.log.txt
: Target gemma_basic not materialized - build required
: Killing previous job: scancel   10733678
: Submitting job: sbatch -o /user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/out/gemma_basic -e /user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/err/gemma_basic    --dependency=afterok:10733695 --parsable /user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/script/gemma_basic >/user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/job/gemma_basic
: Target gemma_basic queued for rebuild


* Exploring the results

** Manhatten plot

Using the functions.R file from the Gemma tutorial for the function to
make a manhatten plot. (This has been modifed to work with 23 chr and
to print the p_wald results).
#+BEGIN_SRC R :file manhatten.png :results output graphics :session *shell*<2> :exports both
library(ggplot2)
library(cowplot)
gwscan <- read.table("./output/taste.assoc.txt", as.is = "rs",header = TRUE)
theme_set(theme_cowplot())
source("../functions.R")
p1 <- plot.gwscan(gwscan)
print(p1)


#+END_SRC

#+RESULTS:
[[file:manhatten.png]]


We can find the actual SNP with the highest association from the
gwas with:

#+BEGIN_SRC R :session *shell*<2> :results output :exports both
i <- which.min(gwscan$p_wald)
gwscan[i,]
#+END_SRC

#+RESULTS:
: i <- which.min(gwscan$p_wald)
: gwscan[i,]
:         chr       rs        ps n_mis n_obs allele1 allele0    af     beta
: 3314306   7 rs713598 141673345   179  3978       G       C 0.419 2.173568
:                 se        p_wald
: 3314306 0.05630622 7.161586e-279


Now we want to do the same with the mgf dossage files.

#+BEGIN_SRC R :session *shell*<2> :results output :exports both

  taste_fam <- read.table('taste.fam')


  sample_file <- read.table('/mnt/storage/private/alspacdata/freeze/original/gi_hrc_g0m_g1/data/bgen/data.sample')

  sample_file_no_head <- sample_file[-c(1,2),]

  merged <- merge(sample_file_no_head, taste_fam, by.x ="V1", by.y="V1",all.x =TRUE)

  pheno_na <- replace(merged$V6.y, merged$V6.y ==-9, 'NA')


  ### set -9 to NA

  write.table(pheno_na, file="pheno.txt",row.names=FALSE, col.names=FALSE,quote=FALSE)


#+END_SRC

We first make the matrix

#+BEGIN_SRC prolog :tangle makeprog
'matrix_full.sXX.txt',{QsubArgs = '--time=72:00:00'} 
<--[],'../../../gemma -bfile ../data -gk 2 -o matrix_full'.

#+END_SRC


Evoke the recipes as submitted jobs on bc4 with slurm.
#+BEGIN_SRC bash :results output

biomake -Q slurm matrix_full.sXX.txt

#+END_SRC

#+RESULTS:
: Target matrix_full.sXX.txt not materialized - build required
: Submitting job: sbatch -o /user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/out/matrix_full.sXX.txt -e /user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/err/matrix_full.sXX.txt   --time=72:00:00  --parsable /user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/script/matrix_full.sXX.txt >/user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/job/matrix_full.sXX.txt
: Target matrix_full.sXX.txt queued for rebuild

Then the code to run an lmm with the relation matrix on chr7

#+BEGIN_SRC prolog :tangle makeprog

bim_bam_path('/mnt/storage/private/alspacdata/freeze/original/omics_freeze_creation/scripts/make_gi_hrc_g0m_g1/bimbam/').

'./output/taste_lmm_bim_bam_chr$(CHR).assoc.txt',{QsubArgs = '--time=36:00:00', bim_bam_path(P)} 
<--[],
'../../../gemma -g $(P)data_$(CHR).mgf -p pheno.txt  -k ./output/matrix_full.sXX.txt -lmm 1 -o taste_lmm_bim_bam_chr$(CHR)'.

'./output/taste_lmm_bim_bam_chr$(CHR).log.txt' <--['./output/taste_lmm_bim_bam_chr$(CHR).assoc.txt'],{true}.



go,{bim_bam_path(P)} <--['./output/taste_lmm_bim_bam_chr07.log.txt'],''.



#+END_SRC
Submit Job to bc4
#+BEGIN_SRC bash :results output

 biomake -Q slurm go

#+END_SRC

#+RESULTS:
: Submitting job: sbatch -o /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/out/taste_lmm_bim_bam_chr07.assoc.txt -e /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/err/taste_lmm_bim_bam_chr07.assoc.txt   --time=36:00:00  --parsable /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/script/taste_lmm_bim_bam_chr07.assoc.txt >/user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/job/taste_lmm_bim_bam_chr07.assoc.txt
: Submitting job: sbatch -o /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/out/taste_lmm_bim_bam_chr07.log.txt -e /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/err/taste_lmm_bim_bam_chr07.log.txt    --dependency=afterok:10734069 --parsable /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/script/taste_lmm_bim_bam_chr07.log.txt >/user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/job/taste_lmm_bim_bam_chr07.log.txt
: Target go not materialized - build required
: Submitting job: sbatch -o /user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/out/go -e /user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/err/go    --dependency=afterok:10734070 --parsable /user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/script/go >/user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/job/go
: Target go queued for rebuild


Plot manhatten for chr7

We need to modify the output file to get a good plot.
#+BEGIN_SRC R :file manhatten_chr7.png :results output graphics :session *shell*<2> :exports both
gwscan_chr7_init <- read.table("./output/taste_lmm_bim_bam_chr07.assoc.txt", as.is = "rs",header = TRUE)

gwscan_chr7_init$chr <- 7

new <- gwscan_chr7_init %>% separate(rs, c("A", "B"),sep=":")

gwscan_chr7_init$ps <-as.numeric(new$B)

#+END_SRC



Using the functions.R file from the Gemma tutorial for the function to
make a manhatten plot. (This has been modifed to work with 23 chr and
to print the p_wald results).
#+BEGIN_SRC R :file manhatten_chr7.png :results output graphics :session *shell*<2> :exports both
library(ggplot2)
library(cowplot)

theme_set(theme_cowplot())
source("../functions.R")
p2 <- plot.region.pvalues(gwscan_chr7_init)
print(p2)


#+END_SRC

#+RESULTS:
[[file:manhatten_chr7.png]]



We can find the actual SNP with the highest association from the LMM
gwas with.

#+BEGIN_SRC R :session *shell*<2> :results output :exports both
i <- which.min(gwscan_chr7_init$p_wald)
gwscan_chr7_init[i,]
#+END_SRC

#+RESULTS:
: i <- which.min(gwscan_chr7_init$p_wald)
: gwscan_chr7_init[i,]
:        chr          rs        ps n_miss                  allele1 allele0    af
: 324722   7 7:111614876 111614876      0 7:111614876:07:111614876       G 0.066
:              beta        se   logl_H1   l_remle      p_wald
: 324722 -0.6467832 0.1339082 -10237.94 0.9150833 1.41415e-06

** 
** 
** 


I dont think the mgf files are formatted correctly for gemma as the
allele1 and allele 0 col in the output are strange and the p_wald is low.


Lets use a modifed version of the mgf file:

#+BEGIN_SRC prolog :tangle makeprog

bim_bam_path('/mnt/storage/private/alspacdata/freeze/original/omics_freeze_creation/scripts/make_gi_hrc_g0m_g1/bimbam/').

'./output/v2_taste_lmm_bim_bam_chr$(CHR).assoc.txt',{QsubArgs = '--time=36:00:00', bim_bam_path(P)} 
<--[],
'../../../gemma -g $(P)mod_data_$(CHR).mgf -p pheno.txt  -k ./output/matrix_full.sXX.txt -lmm 1 -o v2_taste_lmm_bim_bam_chr$(CHR)'.

'./output/v2_taste_lmm_bim_bam_chr$(CHR).log.txt' <--['./output/v2_taste_lmm_bim_bam_chr$(CHR).assoc.txt'],{true}.



go2,{bim_bam_path(P)} <--['./output/v2_taste_lmm_bim_bam_chr07.log.txt'],''.



#+END_SRC
Submit Job to bc4
#+BEGIN_SRC bash :results output

 biomake -Q slurm go2

#+END_SRC

#+RESULTS:
: Submitting job: sbatch -o /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/out/v2_taste_lmm_bim_bam_chr07.assoc.txt -e /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/err/v2_taste_lmm_bim_bam_chr07.assoc.txt   --time=36:00:00  --parsable /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/script/v2_taste_lmm_bim_bam_chr07.assoc.txt >/user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/job/v2_taste_lmm_bim_bam_chr07.assoc.txt
: Submitting job: sbatch -o /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/out/v2_taste_lmm_bim_bam_chr07.log.txt -e /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/err/v2_taste_lmm_bim_bam_chr07.log.txt    --dependency=afterok:10734653 --parsable /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/script/v2_taste_lmm_bim_bam_chr07.log.txt >/user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/job/v2_taste_lmm_bim_bam_chr07.log.txt
: Target go2 not materialized - build required
: Submitting job: sbatch -o /user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/out/go2 -e /user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/err/go2    --dependency=afterok:10734654 --parsable /user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/script/go2 >/user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/job/go2
: Target go2 queued for rebuild


We need to modify the output file to get a good plot.
#+BEGIN_SRC R :session *shell*<2> :results output :exports both
v2_gwscan_chr7 <- read.table("./output/v2_taste_lmm_bim_bam_chr07.assoc.txt", as.is = "rs",header = TRUE)

v2_gwscan_chr7$chr <- 7


v2_gwscan_chr7$ps <-as.numeric(v2_gwscan_chr7$rs)

#+END_SRC

#+RESULTS:
: v2_gwscan_chr7 <- read.table("./output/v2_taste_lmm_bim_bam_chr07.assoc.txt", as.is = "rs",header = TRUE)
: 
: v2_gwscan_chr7$chr <- 7
: 
: 
: v2_gwscan_chr7$ps <-as.numeric(v2_gwscan_chr7$rs)




Using the functions.R file from the Gemma tutorial for the function to
make a manhatten plot. (This has been modifed to work with 23 chr and
to print the p_wald results).
#+BEGIN_SRC R :file v2_manhatten_chr7.png :results output graphics :session *shell*<2> :exports both
library(ggplot2)
library(cowplot)

theme_set(theme_cowplot())
source("../functions.R")
p3 <- plot.region.pvalues(v2_gwscan_chr7)
print(p3)


#+END_SRC

#+RESULTS:
[[file:v2_manhatten_chr7.png]]


We can find the actual SNP with the highest association from the LMM
gwas with.

#+BEGIN_SRC R :session *shell*<2> :results output :exports both
i <- which.min(v2_gwscan_chr7$p_wald)
v2_gwscan_chr7[i,]
#+END_SRC

#+RESULTS:
: i <- which.min(v2_gwscan_chr7$p_wald)
: v2_gwscan_chr7[i,]
:        chr       rs       ps n_miss allele1 allele0    af      beta        se
: 129621   7 36507696 36507696      0       C       T 0.018 -1.213502 0.2667256
:          logl_H1   l_remle       p_wald
: 129621 -10239.26 0.9659095 5.527457e-06


This still has not worked.

Maybe it is the merge function that is causinng the problem.




#+BEGIN_SRC R :session *shell*<2> :results output :exports both

  taste_fam <- read.table('taste.fam')


  sample_file <- read.table('/mnt/storage/private/alspacdata/freeze/original/gi_hrc_g0m_g1/data/bgen/data.sample')

  sample_file_no_head <- sample_file[-c(1,2),]

  joined <- sample_file_no_head %>% left_join(taste_fam, by="V1")

  #merged <- merge(sample_file_no_head, taste_fam, by.x ="V1", by.y="V1",all.x =TRUE)

  pheno_na_j <- replace(joined$V6.y, joined$V6.y ==-9, 'NA')


  ### set -9 to NA

  write.table(pheno_na_j, file="pheno_j.txt",row.names=FALSE, col.names=FALSE,quote=FALSE)


#+END_SRC



Lets use a modifed version of the mgf and new pheno file made by join

#+BEGIN_SRC prolog :tangle makeprog

bim_bam_path('/mnt/storage/private/alspacdata/freeze/original/omics_freeze_creation/scripts/make_gi_hrc_g0m_g1/bimbam/').

'./output/v2_j_taste_lmm_bim_bam_chr$(CHR).assoc.txt',{QsubArgs = '--time=36:00:00', bim_bam_path(P)} 
<--[],
'../../../gemma -g $(P)mod_data_$(CHR).mgf -p pheno_j.txt  -k ./output/matrix_full.sXX.txt -lmm 1 -o v2_j_taste_lmm_bim_bam_chr$(CHR)'.

'./output/v2_j_taste_lmm_bim_bam_chr$(CHR).log.txt' <--['./output/v2_j_taste_lmm_bim_bam_chr$(CHR).assoc.txt'],{true}.



go3<--['./output/v2_j_taste_lmm_bim_bam_chr07.log.txt'],{true}.



#+END_SRC

Submit Job to bc4
#+BEGIN_SRC bash :results output

 biomake -Q slurm go3

#+END_SRC

#+RESULTS:
: Submitting job: sbatch -o /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/out/v2_j_taste_lmm_bim_bam_chr07.assoc.txt -e /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/err/v2_j_taste_lmm_bim_bam_chr07.assoc.txt   --time=36:00:00  --parsable /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/script/v2_j_taste_lmm_bim_bam_chr07.assoc.txt >/user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/job/v2_j_taste_lmm_bim_bam_chr07.assoc.txt
: Submitting job: sbatch -o /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/out/v2_j_taste_lmm_bim_bam_chr07.log.txt -e /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/err/v2_j_taste_lmm_bim_bam_chr07.log.txt    --dependency=afterok:10734904 --parsable /user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/script/v2_j_taste_lmm_bim_bam_chr07.log.txt >/user/work/sn0550/gwas_scar/scripts/notebook/taste/output/.biomake/slurm/job/v2_j_taste_lmm_bim_bam_chr07.log.txt
: Target go3 not materialized - build required
: Killing previous job: scancel   10734903
: Submitting job: sbatch -o /user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/out/go3 -e /user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/err/go3    --dependency=afterok:10734905 --parsable /user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/script/go3 >/user/work/sn0550/gwas_scar/scripts/notebook/taste/.biomake/slurm/job/go3
: Target go3 queued for rebuild

#+BEGIN_SRC R :session *shell*<2> :results output :exports both
v2_j_gwscan_chr7 <- read.table("./output/v2_j_taste_lmm_bim_bam_chr07.assoc.txt", as.is = "rs",header = TRUE)

v2_j_gwscan_chr7$chr <- 7


v2_j_gwscan_chr7$ps <-as.numeric(v2_j_gwscan_chr7$rs)

#+END_SRC

#+RESULTS:
: v2_j_gwscan_chr7 <- read.table("./output/v2_j_taste_lmm_bim_bam_chr07.assoc.txt", as.is = "rs",header = TRUE)
: 
: v2_j_gwscan_chr7$chr <- 7
: 
: 
: v2_j_gwscan_chr7$ps <-as.numeric(v2_j_gwscan_chr7$rs)

Using the functions.R file from the Gemma tutorial for the function to
make a manhatten plot. (This has been modifed to work with 23 chr and
to print the p_wald results).

#+BEGIN_SRC R :file v2_j_manhatten_chr7.png :results output graphics :session *shell*<2> :exports both
library(ggplot2)
library(cowplot)

theme_set(theme_cowplot())
source("../functions.R")
p4 <- plot.region.pvalues(v2_j_gwscan_chr7)
print(p4)


#+END_SRC

#+RESULTS:
[[file:v2_j_manhatten_chr7.png]]

We can find the actual SNP with the highest association from the LMM
gwas with.

#+BEGIN_SRC R :session *shell*<2> :results output :exports both
i <- which.min(v2_j_gwscan_chr7$p_wald)
v2_j_gwscan_chr7[i,]
#+END_SRC

#+RESULTS:
: i <- which.min(v2_j_gwscan_chr7$p_wald)
: v2_j_gwscan_chr7[i,]
:        chr        rs        ps n_miss allele1 allele0    af    beta         se
: 398617   7 141673345 141673345      0       C       G 0.422 2.17343 0.05556671
:          logl_H1    l_remle        p_wald
: 398617 -9619.373 0.03311163 3.330692e-285
