#+TITLE: Genome Wide Association Study of Scar Size in ALSPAC
#+AUTHOR: Sam Neaves

* About

This notebook is written as an [[https://orgmode.org][emacs Org file]] and exported into HTML
and PDF formats. 
Org files are plain text and can be viewed with any plain text file
viewer. 
In addition if the file is opened in [[https://www.gnu.org/software/emacs/][emacs]], interactive features of
running the code in the code blocks can be performed using a feature called
[[https://orgmode.org/worg/org-contrib/babel/][org babel]].
The GIT hub repository for this project is https://github.com/samwalrus/gwas_scar. 
This contains a record of previous versions of the notebook as it has
been developed for transparency. 
Code blocks are in Bash, R and Prolog (only the subset required for
defining makeprog files for[[https://github.com/evoldoers/biomake][ Biomake]] )
The programming style of mixing prose and code blocks in notebooks is
a type of [[https://en.wikipedia.org/wiki/Literate_programming][literate programming]].

This notebook and associated code can be read in a number of ways.
Firstly it can be read as a fixed record of the code that was run and the
results obtained at the time on the hardware used by myself.
Secondly it can be run interactively step by step where a reader is
able to modify code blocks to explore the coded parameter space as well
as add or remove lines of code.
Thirdly it can be read as instructions to run the code as part
of a pipeline.
This can be done by 'tangling' the src_code blocks into source files
which can then be run in a biomake pipeline to  'make' the resulting
figures and output files from the raw data files.

** Initial Setup for interactive code.

The code in this notebook is designed to be open and rerun-able. 
Some aspects of running code on a particular system at a particular
time may need to be modified.
The following instruction give details on how this code was run to
produce this document.

This code was run on two machines, due to how the data and compute are
currently organised in Bristol.
This means that the rcode in this notebook will either be directed at
a local session or a remote session.
Each source code block will be labeled in order with the appropriate
session for this to work.
The code for each session needs to run sequentially due to the
procedural nature of the code.
Running a code block from one session out of sequential order might
lead to unexpected results.

The first part of the code is run on a macbook laptop which is able to
connect to the local SMB drive where ALSPAC phenotype data is stored.
This code extracts the relevant variables and writes these to a file
which is then copied onto the second machine which is part of the
university of Bristol HPC facilities.

To set up the first part of the code to run interactively on the local machine first connect to
the ALSPAC data store using the SMB server (as documented at..).
(If working remotely the VPN will need to be on). 
Then:
1. open this .org file in emacs
2. open a local shell in emacs (M-x shell)
3. Navigate to your local working directory 
4. Start R.

Use the org babel commands to run an execute this code.


The rest of the code is run on the University of Bristol (UOB) high
performance computer (HPC).
As of the summer of 2022, the current HPC machine at UOB is Blue
Crystal Phase 4 (BC4). 
BC4 uses the slurm job scheduler in order to fairly share the compute
resources. 
The compute resources in HPC machines are commonly controlled by login
nodes that then utilize compute nodes.
To run the R code in this note book interactively on BC4 you need to
start an interactive job with slurm to do this you
need to:
1. Open this .org file in emacs.
2. Open a shell (M-x shell) on BC4 over [[https://www.emacswiki.org/emacs/TrampMode][tramp]].
3. Start an interactive slurm job on BC4 [[ijob]]
4. Navigate to your working directory
5. Start R in this shell and run `M-x ess-remote` to connect to the
   ess session.

Make sure the two emacs buffers with R sessions are named as the
session variables in the babel src_code blocks.This notebook assume  *shell* for the local
machine and *shell*<2> for the remote machine.



#+NAME: ijob
#+PROPERTY: header-args :eval never-export
#+BEGIN_SRC bash
srun --nodes=1 --ntasks-per-node=1 --mem=10G --time=04:00:00 --pty bash -i
#+END_src



In order to insert images directly into the org file make sure you
have run the following emacs commands.

#+BEGIN_SRC lisp
(setq org-confirm-babel-evaluate nil)
(add-hook 'org-babel-after-execute-hook 'org-display-inline-images)   
(add-hook 'org-mode-hook 'org-display-inline-images)  

#+END_SRC 

* Aim of the analysis

This analysis is part of the study of genetics of scaring using the
ALSPAC cohort.

** Background

When tissues are damaged in a human any time after birth a complex
wound healing process takes places which results in fibrotic scars. 
These scars can lead to loss of function in the tissue. 
The most severe cases might be skin contraction or organ failure.
For example heart failure after a heart attack.
The precise molecular mechanisms underlying the wound healing process
and scar formation is not well
understood.
This means that there are a lack of effective treatments.

Most of what is known about scar formation has been discovered using
animal models because studying scars in humans is difficult due to to
the unethical nature of delivering standard wounds to people.
The observational studies that have been conducted have mostly been in
burn wounds and these have suggested differences in race suggesting
genetic factors may have an impact.

In this work we attempt to overcome some of the difficulties in human
studies by looking at standard wounds.
This is done by using longitudinal cohort data from the Avon longitudal
parents and children study.
Specifically we look at participants who have had BCG vaccinations
which have resulted in scars.
The BCG vaccination is X.
It often causes a fairly standardized scar due to the routine nature
of the administration of the scar.
In ALSPAC the longest axis of the scar was measured in millimeters, as
well as the age of the scar and the field worker who measured the scar.

* Getting the Phenotype data.

We first need to pull the phenotype and co-var data from the ALSPAC
data store using the R alspac library write this to a file. 
The following code needs to be run in the local R session that is on a
machine that can connect to the 'smb' drive.

The variables that we need to extract are:

1. Sex: kz021 
2. Fieldworker for BCG session: [[https://discovery.closer.ac.uk/item/uk.alspac/586cdfc0-9d9b-42d7-a191-6d54ea64aab5][fm4bc001]] 
3. Participant remembers having BCG injection:[[https://discovery.closer.ac.uk/item/uk.alspac/5541594d-ca18-4b6d-9c91-c5022d8f0063][fm4bc020]]
4. Age participant had BCG (years):[[https://discovery.closer.ac.uk/item/uk.alspac/99c08293-3f88-4bd1-b3e3-5ecf8906f414][fm4bc021]]
5. Bcg scar present:[[https://discovery.closer.ac.uk/item/uk.alspac/32640a73-a2b8-4f96-9bab-fa0c31d0fdf6][fm4bc022]]
6. Size of scar (mm):[[https://discovery.closer.ac.uk/item/uk.alspac/dbe1bfce-67e5-4fd8-9e48-6bd3e15a3da4][fm4bc023]]
7. Description of scar:[[https://discovery.closer.ac.uk/item/uk.alspac/d80de95d-0d92-474e-a6dd-27034a1db566][fm4bc024]]
8. Color of scar:[[https://discovery.closer.ac.uk/item/uk.alspac/b130b42c-6100-4d5a-8b25-d98ed24a4a8d][fm4bc025]]



*Local R session*
#+NAME: get_vars
#+BEGIN_SRC R :session *shell* :results output :tangle get_pheno.R
  library(devtools)
  ### library(alspac is available from: https://github.com/explodecomputer/alspac)
  library(alspac)
  setDataDir("/Volumes/ALSPAC-Data/")
  data(current)
  data(useful)

  ### Sex
  ### kz021

  ### Fieldworker for BCG session: FOM4
  ### fm4bc001

  ### Participant remembers having BCG injection: FOM4
  ### fm4bc020

  ### Age participant had BCG (years): FOM4
  ### fm4bc021

  ### Bcg scar present
  ### fm4bc022

  ### size of scar (mm)
  #### fm4bc023

  ### Description of scar
  ### fm4bc024

  ### Color of scar
  ### fm4bc025

  varnames <- c("kz021","fm4bc001","fm4bc020","fm4bc021","fm4bc022","fm4bc023", "fm4bc024","fm4bc025")
  vars <- findVars(varnames)
  results <- extractVars(vars)

  write.csv(as.data.frame(results),file="./pheno1.csv")

#+END_SRC

#+RESULTS: get_vars
#+begin_example
Loading required package: usethis
Loading required package: readstata13
Loading required package: haven
Loading required package: stringr
Loading required package: plyr
Loading required package: dplyr

Attaching package: ‘dplyr’

The following objects are masked from ‘package:plyr’:

    arrange, count, desc, failwith, id, mutate, rename, summarise,
    summarize

The following objects are masked from ‘package:stats’:

    filter, lag

The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union

Loading required package: parallel
R/alspac version 0.33.0
We have made several changes to how the extractVars function works.
This may lead to slightly different formatting compared to previously.
See also the new createDictionary function.
Warning message:
package ‘dplyr’ was built under R version 4.0.5
Warning message:
In findVars(varnames) :
  One or more variables have the same name (fix with filterVars()): kz021
Starting extraction from 3 files in the ALSPAC data directory
Extracting from: /Volumes/ALSPAC-Data//Current/Other/Cohort Profile//cp_2b.dta
Extracting from: /Volumes/ALSPAC-Data//Current/Clinic/Child//f08_4d.dta
Extracting from: /Volumes/ALSPAC-Data//Current/Clinic/Adult//FOM4_3b.dta
Collapsing data
New names:
kz021...4
kz021...6
Starting extraction from 7 files in the ALSPAC data directory
Extracting from: /Volumes/ALSPAC-Data//Current/Quest/Mother//a_3e.dta
Extracting from: /Volumes/ALSPAC-Data//Current/Quest/Mother//b_4f.dta
Extracting from: /Volumes/ALSPAC-Data//Useful_data/bestgest//bestgest.dta
Extracting from: /Volumes/ALSPAC-Data//Current/Quest/Mother//c_8a.dta
Extracting from: /Volumes/ALSPAC-Data//Current/Other/Cohort Profile//cp_2b.dta
Extracting from: /Volumes/ALSPAC-Data//Current/Other/Sample Definition//kz_5c.dta
Extracting from: /Volumes/ALSPAC-Data//Current/Other/Sample Definition//mz_5a.dta
Collapsing data
New names:
kz021...26
Automatically removing data for individuals who have withdrawn consent.
#+end_example



This file then needs to be moved to the BC4 working directory

#+NAME: move_pheno
#+PROPERTY: header-args :eval never-export
#+BEGIN_SRC bash
mv $(LocalPath)/$(Phenofile) $(HPCPath)/$(Phenofile)
#+END_src

* Exploring Phenotype data

** Reading the file into R

#+BEGIN_SRC R :session *shell*<2> :results output :exports both

  raw_pheno <-read.csv('../../alspac_pheno_raw/pheno1.csv')


#+END_SRC

#+RESULTS:
: 
: raw_pheno <-read.csv('../../alspac_pheno_raw/pheno1.csv')

** Excluding people
Participants without scars are excluded.

#+BEGIN_SRC R :session *shell*<2> :results output :exports both
  library(dplyr)
  dim(raw_pheno)

  pheno_with_scar <-raw_pheno %>% filter(fm4bc022==1)

  dim(pheno_with_scar)
#+END_SRC

#+RESULTS:
: library(dplyr)
: dim(raw_pheno)
: [1] 15645    40
: 
: pheno_with_scar <-raw_pheno %>% filter(fm4bc022==1)
: 
: dim(pheno_with_scar)
: [1] 971  40

*** TODO Why is the qlet A for all of this data? (Map to M?)

** Examining scar size 
*** summary

#+BEGIN_SRC R :session *shell*<2> :results output :exports both
  summary(pheno_with_scar$fm4bc023)
#+END_SRC

#+RESULTS:
: summary(pheno_with_scar$fm4bc023)
:    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
:  -1.000   4.000   5.000   6.313   8.000  35.000


*** plot

#+BEGIN_SRC R :file scar_size_raw_plot.png :results output graphics :session *shell*<2> :exports both
library(ggplot2)
library(cowplot)

theme_set(theme_cowplot())
p1 <- ggplot(pheno_with_scar,aes(fm4bc023)) +
geom_histogram(fill = "darkblue",
na.rm = TRUE)
print(p1)



#+END_SRC

#+RESULTS:
[[file:scar_size_raw_plot.png]]

*** inverse rank transform summary and plot

We inverse rank transform the scar measurement residuals because this
results in a normally distributed quantitive trait.

#+BEGIN_SRC R :session *shell*<2> :results output :exports both
  #install.packages("RNOmni")
  library(RNOmni)

  pheno_with_scar$scar_size_rank_transform <-RankNorm(pheno_with_scar$fm4bc023)

  summary(pheno_with_scar$scar_size_rank_transform)
#+END_SRC

#+RESULTS:
: #install.packages("RNOmni")
: library(RNOmni)
: 
: pheno_with_scar$scar_size_rank_transform <-RankNorm(pheno_with_scar$fm4bc023)
: 
: summary(pheno_with_scar$scar_size_rank_transform)
:      Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
: -3.218863 -0.739737 -0.192157  0.006134  0.707877  3.218863



(Using the package from: Operating characteristics of the rank-based inverse normal transformation for quantitative trait analysis in genome-wide association studies)

#+BEGIN_SRC R :file scar_size_inverse_rank_transform_plot.png :results output graphics :session *shell*<2> :exports both
library(ggplot2)
library(cowplot)

theme_set(theme_cowplot())
p1 <- ggplot(pheno_with_scar,aes(scar_size_rank_transform)) +
geom_histogram(fill = "darkblue",
na.rm = TRUE)
print(p1)

#+END_SRC

#+RESULTS:
[[file:scar_size_inverse_rank_transform_plot.png]]


*** Writing to file

We need to write two files. 
One for our phenotype and one for our covar.
They need to be in the same order as our genotype files.
The covar file needs a collumn of ones for the intercept.

#+BEGIN_SRC R :session *shell*<2> :results output :exports both

  head(pheno_with_scar)

  fam_original <- read.table("combined.fam")

  head(fam_original)

  new_fam_all <- merge(fam_original, pheno_with_scar, by.x="V1", by.y="alnqlet", all.x = TRUE)

  new_fam <- new_fam_all[c("V1","V2","V3","V4","V5","scar_size_rank_transform")]

  new_fam[is.na(new_fam)] <- -9

  summary(new_fam[,c("scar_size_rank_transform")])

  write.table(new_fam,"scar.fam", row.names=FALSE, col.names=FALSE, quote=FALSE)

  #write.table(pheno_with_scar,""row.names =FALSE, col.names =FALSE)
#+END_SRC

* Genetic data

** HRC data

We want to use the HRC data.
These need to be in the correct format for Gemma.

Simon used the bimbam mean genotype file format (mgf).
We are initially using plink format files.

*** TODO todo check the plink file format of hrc is best guess or dossage?






* Running GWASes with Gemma

Todo this we are going to tangle a biomake file with rules for how to
execute Gemma, including the dependencies. The command for this is C-c C-v t.

We can then execute the recipes in the make file over slurm in compute
jobs.


** Aim compute the p-values to assess support for association between scar length and SNPs on chromosomes 1–22.

1. Run a basic association analysis in GEMMA.
2. Run an LMM-based association analysis in GEMMA.
3. Compare the two association analyses.

** Basic association

Create the rule:

#+PROPERTY: header-args :eval never-export
#+BEGIN_SRC prolog :tangle makeprog
hrc_plink_path(P):-
  consult('./paths.pl'),
  path('hrc_plink',P).

'./output/scar.assoc.txt',{QsubArgs = '--time=18:00:00'}<--
['scar.bed','scar.bim','scar.fam'],'../../gemma -b scar -lm 1 -outdir . -o ./output/scar'.

'./output/scar.log.txt' <--['./output/scar.assoc.txt'],{true}.

[gemma_basic] <--['./output/scar.log.txt'].
#+END_SRC

We can evoke a dry run of the recipes to check the commands 

#+BEGIN_SRC bash 

biomake -n gemma_basic

#+END_SRC

#+RESULTS:
| ../../gemma | -b          | scar | -lm          | 1 | -outdir |        0 | -o | ./output/scar |
| Target      | gemma_basic | not  | materialized | 0 | build   | required |    |               |


Evoke the recipes as submitted jobs on bc4 with slurm.
#+BEGIN_SRC bash 

biomake -Q slurm gemma_basic

#+END_SRC

#+BEGIN_SRC bash 

biomake -n gemma_basic

#+END_SRC

#+BEGIN_SRC bash 

biomake test.txt

#+END_SRC

** LMM association

* Exploring the results

** Manhatten plot


#+BEGIN_SRC R :file manhatten.png :results output graphics :session *shell*<2> :exports both
library(ggplot2)
library(cowplot)

print(p1)

#+END_SRC

** Zoom in on chr and region

#+BEGIN_SRC R :file zoom.png :results output graphics :session *shell*<2> :exports both
library(ggplot2)
library(cowplot)

print(p1)

#+END_SRC


** Snp with highest association

** load genotype

** box plot

#+BEGIN_SRC R :file box_plot.png :results output graphics :session *shell*<2> :exports both
library(ggplot2)
library(cowplot)

print(p1)

#+END_SRC


** Quantify relationship with linear model
